addons/survey/models/survey_user_input.py

```diff
@@ -230,6 +230,8 @@ def _mark_done(self):
 
     def get_start_url(self):
         self.ensure_one()
+        if (self.survey_id.access_mode == 'public'):
+            return self.survey_id.get_start_url();
```

Comment:
no ; in python (it works but you should not to use the same style as rest of code)

```suggestion
            return self.survey_id.get_start_url()
```